### PR TITLE
Zack/greedy contact set loss

### PR DESCRIPTION
**Pull request by znichols**
## Summary

- adds contacts-v1 greedy latent contact-set loss implementation and tests
- adds exp156 training/evaluation launcher with stock next-token vs greedy-set arms
- records current H100/GB200 smoke-run findings and caveats in the experiment docs
- adds greedy-set validation hooks, optional auxiliary next-token validation for greedy runs, and configurable `EXP156_LEARNING_RATE`

## Current status

- 1×H100 short greedy eval succeeded with both greedy-set and next-token validation losses.
- 4×GB200 next-token and greedy runs completed; high-LR runs looked unstable/overfit.
- Multi-GPU greedy next-token eval is disabled by default for current runs via `EXP156_ENABLE_GREEDY_NEXT_TOKEN_EVAL=0` because Levanter's fused CE path can hit an axis-size mismatch on local multi-GPU meshes.
- Follow-up 8×H100 lower-LR runs are in flight outside this PR.

## Testing

- `python3 -m py_compile experiments/exp156_models_contacts_v1_greedy_set_loss/train.py`
- `uv run --project marinfold pytest marinfold/tests/document_structures/contacts_v1/test_greedy_set_loss.py`

Note: `uv run --project experiments/exp156_models_contacts_v1_greedy_set_loss ...` failed on macOS while resolving CUDA/NVIDIA wheels for the GPU extra; used direct `python3 -m py_compile` for syntax validation instead.

### CODE REVIEW

**Comment by znichols** [comment]
abandoning as per discussion with tim
